### PR TITLE
add amazon data source

### PR DESCRIPTION
**Pull request by rhagarty**
This adds ability to run either IBM data store data or Amazon.com data. Commenting/uncommenting out the appropriate discovery values in .env will determine which one to use.

### CODE REVIEW

File: ?
Comment (markstur) [inline ?:1]: Add license

Comment (markstur) [inline ?:1]: I would inline items

Comment (markstur) [inline ?:1]: one possibility for git rid of all the if x in y, for i in y[x] is to instead do
for x in y.get(x, []):
I find it a little annoying either way, but sometime it's nice to have less indenting conditions, etc...

Comment (markstur) [inline ?:1]: this change is unrelated. These lines were moved around recently so probably a merge issue.  Also the workspace line will change soon with my PR 🤞

Comment (scottdangelo) [inline ?:1]: "data_source" -> "data_source:"

Comment (scottdangelo) [inline ?:1]: combine 265 and 267:
elif:

Comment (scottdangelo) [reply ?:1]: We might want to construct it so that it's something like: 
if "amazon":
   # deal with amazon data
elif "ibm_store":
   #deal with ibm_store
#elif <your_dev_stuff>:
   #deal with your custom dev stuff

Comment (scottdangelo) [inline ?:1]: We should probably call out that this is for ibm_store, ie.:
elif data_source == "ibm_store"

Comment (scottdangelo) [inline ?:1]: same

**Review by scottdangelo — CHANGES_REQUESTED** [review]
I haven't tested yet, but here's a couple things...

File: ?
Comment (scottdangelo) [inline ?:1]: If no data_source is set there's no results, so we probably should log an error.

Comment (scottdangelo) [inline ?:1]: Docstring should discuss the curated data_source 's for ibm_store and amazon, that a user could add something different in the env, etc.
We should be clear on things that are hard-coded and would need to change when a dev uses a different data_source, which we assume they eventually will when they move beyond testing.

Comment (scottdangelo) [inline ?:1]: Same as above. This code is specific to the data_source.

**Review by scottdangelo — CHANGES_REQUESTED** [review]
This kinda works, but results are spotty. I think we should discuss and maybe include Michael to see how good this has to be.

File: ?
Comment (rhagarty) [reply ?:1]: I will do that, but I think the best solution is to throw an exception at startup, just like if they fail to provide proper usernames/passwords.

**Review by scottdangelo — APPROVED** [review]
